### PR TITLE
Update display hook to deal with ; in comment

### PR DESCRIPTION
**Pull request by Carreau**
PR #8321 and #8457 combined + tab -> space. 

(will do more fixes if Travis fails)

### CODE REVIEW

File: IPython/core/displayhook.py [hunk 1]
```diff
@@ -83,9 +84,19 @@ def check_for_underscore(self):
     def quiet(self):
         """Should we silence the display hook because of ';'?"""
         # do not print output if input ends in ';'
+
+        cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
```
Comment (takluyver) [inline IPython/core/displayhook.py]: This line needs to be inside the try/except IndexError. The token loop doesn't.

**Comment by Carreau** [comment]
Rebased and clean. Waiting for tests.

**Comment by Carreau** [comment]
Will close #8002 also. Might need manual closing.

**Comment by takluyver** [comment]
Tests passing. In we go!

**Comment by Carreau** [comment]
Oh, that was quick ! Thanks !

**Pull request merged**